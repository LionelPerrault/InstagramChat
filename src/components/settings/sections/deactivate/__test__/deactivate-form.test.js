import React from 'react'
import { create } from 'react-test-renderer'
import DeactivateForm from '../deactivate-form'
import { shallow } from 'enzyme'

describe('DeactivateForm Component', () => {
  const comp = <DeactivateForm password="mypassword" change={jest.fn()} />

  it('should match snapshot', () => {
    const tree = create(comp).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should show <Prompt/> when form is submitted', () => {
    const wrapper = shallow(comp)
    wrapper
      .find('form.dlt_acc_form')
      .simulate('submit', { preventDefault() {} })
    expect(wrapper.find('Prompt').exists()).toBe(true)
  })

  it("should mock deactivate action when clicked on Prompt's done button", () => {
    const wrapper = shallow(comp)
    wrapper.setState({ showPrompt: true })
    wrapper.find('Prompt').prop('action')({
      preventDefault() {},
    })
  })
})
