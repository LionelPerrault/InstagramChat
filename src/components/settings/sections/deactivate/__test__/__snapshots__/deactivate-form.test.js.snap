// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeactivateForm Component should match snapshot 1`] = `
<form
  className="dlt_acc_form"
  onSubmit={[Function]}
>
  <input
    autoComplete="false"
    autoFocus={true}
    disabled={false}
    maxLength="255"
    onChange={[MockFunction]}
    placeholder="Your password.."
    required={true}
    spellCheck="false"
    type="password"
    value="mypassword"
  />
  <input
    disabled={false}
    type="submit"
    value="Deactivate"
  />
</form>
`;
